PROTO_tests/tests/symbol_table_test.py

```diff
@@ -0,0 +1,766 @@
+import pytest
+
+import arkouda as ak
+from arkouda.pdarrayclass import RegistrationError
+
+N = 100
+UNIQUE = N // 4
+
+DTYPES = [ak.int64, ak.uint64, ak.bool, ak.float64, ak.bigint, ak.str_]
+
+
+def clean_registry():
+    ak.clear()
+    for registered_name in ak.list_registry()["Objects"]:
+        ak.unregister(registered_name)
+    ak.clear()
+
+
+@pytest.fixture(autouse=True)
+def cleanup():
+    clean_registry()
+    yield
+    clean_registry()
+
+
+class TestRegistration:
+    @staticmethod
+    def make_pdarray(dtype, size):
+        if dtype in [ak.int64, ak.uint64]:
+            return ak.arange(size)
+        elif dtype == ak.bigint:
+            return ak.arange(size) + 2**200
+        elif dtype == ak.bool:
+            return ak.randint(0, 2, size, dtype=ak.bool)
+        elif dtype == ak.float64:
+            return ak.linspace(-2.5, 2.5, size)
+        elif dtype == ak.str_:
+            return ak.random_strings_uniform(0, 2, size)
+
+    @pytest.mark.parametrize("dtype", DTYPES)
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_pdarray_registration(self, dtype, size):
+        reg_name = "MyPDArray"
+        pda = self.make_pdarray(dtype, size)
+        pda.register(reg_name)
+
+        # validate registered_name set
+        assert pda.registered_name == reg_name
+        assert pda.is_registered()
+
+        reg = ak.list_registry()
+
+        # assert that the object is registered
+        assert reg_name in reg["Objects"]
+        # assert that the sym entry name is recorded
+        assert pda.name in reg["Components"]
+
+        # validate attach
+        attach_pda = ak.attach(reg_name)
+        assert attach_pda.is_registered()
+        assert pda.registered_name == attach_pda.registered_name
+        assert pda.to_list() == attach_pda.to_list()
+        if dtype == ak.str_:
+            assert isinstance(attach_pda, ak.Strings)
+        else:
+            assert isinstance(attach_pda, ak.pdarray)
+
```

Comment:
Could one-line this:
`        assert isinstance(attach_pda, ak.Strings if dtype == ak.str_ else ak.pdarray)
`

Reply:
Yes, it can be written this way. There is no performance advantage or functionality alteration with doing so though. For right now since we need to get these tests merged lets leave it as is.